app/models/data_model/interaction_claim.rb

```diff
@@ -13,5 +13,17 @@ class InteractionClaim < ::ActiveRecord::Base
     def self.for_show
       eager_load(:interaction_claim_types, :interaction_claim_attributes, :source, :drug_claim, gene_claim: [:gene, :source])
     end
+
+    def self.for_tsv
+      includes(:source, :interaction_claim_types, drug_claim: [:drug], gene_claim: [:gene])
```

Comment:
@acoffman @ahwagner Is it better here to use `eager_load` instead of `includes` here? I'm not exactly sure in which case to use one over the other.

Reply:
I took a look at it with `Benchmark.measure`, and it seems to me like `includes` works well for this purpose. Looking at the Rails source code for the two methods ([includes](https://apidock.com/rails/v4.0.2/ActiveRecord/QueryMethods/includes%21), [eager_load](https://apidock.com/rails/v4.0.2/ActiveRecord/QueryMethods/eager_load%21)), it seems to me that eager_load does slightly less compute (no check for blank, no calls to flatten or uniq), but returns cleaner results. 

However, I may be missing some ActiveRecord magic here, as [this post](http://blog.scoutapp.com/articles/2017/01/24/activerecord-includes-vs-joins-vs-preload-vs-eager_load-when-and-where) states that `eager_load` makes fewer queries (1 vs 2 for `includes`), and that is reflected in the call to `for_tsv`. However, since `for_tsv` is only getting called once per export operation, that distinction is moot--both ops performed in ~25-30ms. For this reason, I would stick with `includes`, as it may do a better job of cleaning up the results, and it matches the recommendations in the [Rails Guides](http://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations).

@acoffman, feel free to correct if I'm off base here.